src/ipc/dbus/DBusObjectRegistry.cpp

```diff
@@ -78,30 +89,52 @@ void DBusObjectRegistry::unregisterObject(const QString &objectPath)
 {
     init();
     auto serviceName = DBusManager::instance().serviceName();
-    if (m_master) {
+    if (isMaster()) {
         m_master->unregisterObject(objectPath, serviceName);
     } else {
         return m_objectRegistryAsyncProxy->unregisterObject(objectPath, serviceName);
     }
 }
 
-const QMap<QString, QString> &DBusObjectRegistry::objects(bool blocking)
+QMap<QString, QString> DBusObjectRegistry::objects(bool blocking)
 {
     init();
-    if (m_master) {
-        return m_master->objects();
-    } else {
-        if (blocking) {
+    if (!isMaster() && blocking) {
+        if (m_objects.isEmpty()) {
             if (m_objectRegistryProxy == nullptr) {
-                m_objectRegistryProxy = new facelift::ipc::dbus::ObjectRegistryIPCDBusProxy(this);
+                m_objectRegistryProxy = new ObjectRegistryIPCDBusProxy(this);
                 m_objectRegistryProxy->ipc()->setServiceName(m_serviceName);
                 m_objectRegistryProxy->connectToServer();
             }
-            return m_objectRegistryProxy->objects();
-        } else {
-            return m_objectRegistryAsyncProxy->objects();
+            return m_objectRegistryProxy->getObjects();
```

Comment:
This is going to generate multiple blocking IPC calls during the application startup, transferring a potentially long registry.